### ISSUE TITLE
Cursor jumps back and forward on switching

### ISSUE THREAD

**Issue by SirIntellegence** [symless/synergy-core#5992]
### Operating Systems ###

Server: Windows 10 ver 1703
Client: OS X 10.10.5

### Synergy Version ###

1.8.8-stable-c30301e (win)
1.8.8 (that is all I could get for the mac)

### Steps to reproduce bug ###

I try to move my mouse from my windows server to my Mac client and my mouse jumps back to the windows server several times before I can get it to the Mac. I think it is related to the clipboard, but it happens even if I have a small amount of text on the clipboard.

### Other info ###

* When did the problem start to occur? Don't know...
* Is there a way to work around it? Not that I know of.
* Does this bug prevent you from using Synergy entirely? No

Server Log: https://pastebin.com/ytwGe4gm
The Client log is a bunch of:
[2017-04-11T11:13:24] WARNING: cursor may not be visible
[2017-04-11T11:13:24] INFO: entering screen
[2017-04-11T11:13:24] DEBUG: showing cursor
[2017-04-11T11:13:24] WARNING: cursor may not be visible
[2017-04-11T11:13:25] INFO: leaving screen
[2017-04-11T11:13:25] DEBUG: hiding cursor
[2017-04-11T11:13:25] WARNING: cursor may not be visible

**Comment by XinyuHou** [comment]
@SirIntellegence 
Thank you for all the details.

The server log doesn't seem to cover the time that issue happens. Is this correct?

**Comment by SirIntellegence** [comment]
No. It does cover the the time the issue happened. If it helps, I can make
a video with the issue happening with the log visible. I think it is
related to the bogus Delta motion.

**Comment by XinyuHou** [comment]
@SirIntellegence 
Sorry my bad. Do you have multiple monitors on your server side?

**Comment by SirIntellegence** [comment]
Yes, I do. Would that break it? I also have multiple monitors on the client as well.

**Comment by XinyuHou** [comment]
@SirIntellegence 
Could try to experiment to set the primary monitor on the left most or any other positions to see that makes any difference? Thanks.

**Comment by SirIntellegence** [comment]
Changing my primary monitor from the bottom left to the bottom right (of a square of 4) is much better, but it sill does it.

**Comment by XinyuHou** [comment]
> Changing my primary monitor from the bottom left to the bottom right (of a square of 4)

Could you give us a bit more information on this? Thanks.

**Comment by SirIntellegence** [comment]
![2017-04-21 12_35_05-settings](https://cloud.githubusercontent.com/assets/3401182/25291412/221252f8-268f-11e7-8fdd-759cd7bdbce6.png)
Config: https://pastebin.com/GHAEfjwy
Does that Help?

**Comment by SirIntellegence** [comment]
The wide ones are 1440x900, the others are 1280x1024

**Comment by SirIntellegence** [comment]
If it helps, "my-slave" is a Linux computer, and "my-laptop" hasn't been connected for a while. The mac has 2 screens of 1440x900, and the Linux computer has 1 screen of 1440X900

**Comment by XinyuHou** [comment]
@SirIntellegence 
The 4 screens you showed, are they from one machine?

Based on your configuration file, does the setup looks like this?
![image](https://cloud.githubusercontent.com/assets/9194706/25536446/c2f19212-2c32-11e7-9308-0a61655c45d4.png)

**Comment by SirIntellegence** [comment]
Yes. That is correct.

**Comment by SirIntellegence** [comment]
Have you made any progress on this? Sometimes, it takes more then 10 tries to get to the other computer.

**Comment by XinyuHou** [comment]
@SirIntellegence 
> I try to move my mouse from my windows server to my Mac client and my mouse jumps back to the windows server several times before I can get it to the Mac.on issue.  

Does the cursor jump back to the center of the server or where you enter the client?

**Comment by SirIntellegence** [comment]
It usually jumps to the center of the bottom left or the center in between the two bottom screens (or further to the right). I have noticed that the return to the main computer sometimes occurs after the cursor is on the client (but just barely). I think it also happens less often if I am moving the cursor across the boundary more slowly.

**Comment by XinyuHou** [comment]
Please try to enable relative mouse movement, which is in server configure->advanced settings

**Comment by SirIntellegence** [comment]
It still breaks, probably even more reliably. Here is a [debug2 log](https://pastebin.com/xURifepr). I typed "test" after the cursor went to and stayed on the client computer. This happened twice in that log file.

**Comment by SirIntellegence** [comment]
What is the status of this? It still happens for me.

**Comment by Dixeran** [comment]
Maybe you can try this:
![screenshot_2017-06-07-20-31-59](https://user-images.githubusercontent.com/22679810/26889469-dfa05e5c-4be0-11e7-89a5-c8a03f54e287.png)
I was facing the same problem with you, the cursor just jumped back when I moved it to the border, but after I try the same way in image, the problem gone away.

**Comment by SirIntellegence** [comment]
Sadly, that does not fix the issue. All of my screens are at 100% dpi

**Comment by SirIntellegence** [comment]
Here is a zip with DEBUG2 logs from the client and the server along with my new config (I added two screens to the mac, so its monitor arrangement is the same as the server). "test" was typed after a successful switch. I eventually decided to add one '!' after "test" to indicate one try, so "test!!" indicates that the switch succeeded on the second attempt. If there is anything else I can do to help you figure this out, please let me know.
[SynergyLogs.zip](https://github.com/symless/synergy/files/1072035/SynergyLogs.zip)

**Comment by SirIntellegence** [comment]
Note: the entries for the imac and laptop are commented out, so you can ignore them

**Comment by SirIntellegence** [comment]
It appears I didn't comment out the entries for the laptop... I just did on my computer and it sill has issues. The laptop hasn't been connected to the Synergy server for months.

**Comment by SirIntellegence** [comment]
Note: I have posted a bounty on this bug (https://www.bountysource.com/issues/44030981-cursor-jumps-back-and-forward-several-times-on-switching)

**Comment by frostyfrog** [comment]
I just wanted to state that I'm running into this issue as well, however, I only see it when I move my mouse to the left of my main monitor. I've tried it when the client has been either linux or windows. I've also tried it with an older synergy client (back from the days when synergy was asking for donations and wasn't a paid solution) and the latest synergy client. Seeing as the issue occurs in all scenarios (with the server being the latest stable release), I'd hazard a guess that it's a server issue.

Below is my modest config file. Frostyfrog is the server here and Think is the client. Both have 2 monitors of unique proportions, however, the right Think and left Frostyfrog displays are the same at 1920x1080. Both computers have their DPI settings set to 100%. Switching it from relative to absolute makes the issue less noticeable, but it is still there. An idea of how noticeable it is when relative mouse movement is on? I once tried for 5 seconds to get my mouse to the other screen. I had to crank up the DPI on my mouse in the end to "blow past the invisible wall", so to speak.
```
section: screens
	Frostyfrog:
		switchCorners = all
		switchCornerSize = 20
	Think:
		switchCorners = all
		switchCornerSize = 20
end
section: links
	Frostyfrog:
		left = Think
		right = Think
	Think:
		left = Frostyfrog
		right = Frostyfrog
end
section: aliases
	Frostyfrog:
		Frostyfrog-pc
		London
		DESKTOP-I5IQT3Q
	Think:
		Colton-THINK
		Ogre
end
section: options
	relativeMouseMoves = true
	screenSaverSync = false
	win32KeepForeground = true
end
```

**Comment by frostyfrog** [comment]
Oh, since logs were provided earlier, here is part of my server logs when I tried moving my mouse cursor between monitors (with absolute mouse movement):
```
[2017-07-02T19:50:23] INFO: switch from "Think" to "Frostyfrog" at 1,601
[2017-07-02T19:50:23] INFO: entering screen
[2017-07-02T19:50:23] INFO: switch from "Frostyfrog" to "Think" at 3519,525
[2017-07-02T19:50:23] INFO: leaving screen
[2017-07-02T19:50:23] INFO: switch from "Think" to "Frostyfrog" at 3,601
[2017-07-02T19:50:23] INFO: entering screen
[2017-07-02T19:50:23] INFO: switch from "Frostyfrog" to "Think" at 3519,524
[2017-07-02T19:50:23] INFO: leaving screen
[2017-07-02T19:50:23] INFO: switch from "Think" to "Frostyfrog" at 1,601
[2017-07-02T19:50:23] INFO: entering screen
[2017-07-02T19:50:23] INFO: switch from "Frostyfrog" to "Think" at 3519,523
[2017-07-02T19:50:23] INFO: leaving screen
```

**Comment by SirIntellegence** [comment]
It might help them more if you get them a level of Debug logging. Then they can see things like "dropped bogus delta motion" and such. A debug2 log might help the most, just remember that it logs keystrokes on that level. That way, they can walk through the logic and see what is going on.

**Comment by h24821807** [comment]
Hi,

I have the same problem. What I notice is if you move your mouse very quickly to the client, it tends to switch back to server. But if you move your mouse slowly, you will get a successful switch.

**Comment by frostyfrog** [comment]
I've been meaning to get more data to this. I don't have access to the logs
since I'm at work, however, just an comment to the last commentor's
message. Yes, it doesn't appear to happen [as often] when the mouse is
moving slowly. However, I have seen it happen even when I'm moving my mouse
slowly and it's set to 800 DPI. I still see it happen, it just bounces back
a handful of times instead of 20-60 times.

**Comment by SirIntellegence** [comment]
That happens with me too. If I go slow, it works better. If I am not going slow, I get the "dropped bogus delta motion" stuff in the log, even when it works.

**Comment by SirIntellegence** [comment]
Another thing I have noticed is that it usually has issues when moving to a computer on the left, hardly on the right. Is this happening to you guys as well?

**Comment by frostyfrog** [comment]
That appears to be correct, from what I remember. Then again, my client is
on the left and my server is on the right, so I assumed that it was because
I was moving from my server to a client. (using an older client didn't
help, as the issue still persisted when using a months/year old client)

~ *Frostyfrog* ^.^

**Comment by jonathansampson** [comment]
This is happening to me too; if I go slowly, it appears to work.

**Comment by SirIntellegence** [comment]
This is odd. I did a "clean" install of Windows and I haven't noticed the issue since...

**Comment by jonathansampson** [comment]
@SirIntellegence I too have not noticed it in a while.

**Comment by primehalo** [comment]
I still have this happen all the time using 1.9.1-stable-2a0225c1 for both server (Windows 10 with latest updates) and client (Mac OSX 10.13.5). I have to move the mouse slowly to switch otherwise it just bounces off the edge of the screen:

```
[2018-06-17T12:44:17] INFO: switch from "MY-PC1" to "192.168.1.101" at 1919,655
[2018-06-17T12:44:17] INFO: leaving screen
[2018-06-17T12:44:17] DEBUG: open clipboard
[2018-06-17T12:44:17] DEBUG: close clipboard
[2018-06-17T12:44:17] DEBUG: ignored screen "MY-PC1" update of clipboard 0 (unchanged)
[2018-06-17T12:44:17] DEBUG: open clipboard
[2018-06-17T12:44:17] DEBUG: close clipboard
[2018-06-17T12:44:17] DEBUG: ignored screen "MY-PC1" update of clipboard 1 (unchanged)
[2018-06-17T12:44:17] DEBUG: dropped bogus delta motion: -2973,+55
[2018-06-17T12:44:17] DEBUG: dropped bogus delta motion: -2890,+55
[2018-06-17T12:44:17] INFO: switch from "192.168.1.101" to "MY-PC1" at -1838,655
[2018-06-17T12:44:17] INFO: entering screen
[2018-06-17T12:44:17] INFO: switch from "MY-PC1" to "192.168.1.101" at 1919,656
[2018-06-17T12:44:17] INFO: leaving screen
[2018-06-17T12:44:17] DEBUG: open clipboard
[2018-06-17T12:44:17] DEBUG: close clipboard
[2018-06-17T12:44:17] DEBUG: ignored screen "MY-PC1" update of clipboard 0 (unchanged)
[2018-06-17T12:44:17] DEBUG: open clipboard
[2018-06-17T12:44:17] DEBUG: close clipboard
[2018-06-17T12:44:17] DEBUG: ignored screen "MY-PC1" update of clipboard 1 (unchanged)
[2018-06-17T12:44:17] DEBUG: dropped bogus delta motion: -2897,+60
[2018-06-17T12:44:17] INFO: switch from "192.168.1.101" to "MY-PC1" at -1908,653
[2018-06-17T12:44:17] INFO: entering screen
[2018-06-17T12:44:17] INFO: switch from "MY-PC1" to "192.168.1.101" at 1919,657
[2018-06-17T12:44:17] INFO: leaving screen
[2018-06-17T12:44:18] DEBUG: open clipboard
[2018-06-17T12:44:18] DEBUG: close clipboard
[2018-06-17T12:44:18] DEBUG: ignored screen "MY-PC1" update of clipboard 0 (unchanged)
[2018-06-17T12:44:18] DEBUG: open clipboard
[2018-06-17T12:44:18] DEBUG: close clipboard
[2018-06-17T12:44:18] DEBUG: ignored screen "MY-PC1" update of clipboard 1 (unchanged)
[2018-06-17T12:44:18] DEBUG: dropped bogus delta motion: -2894,+61
[2018-06-17T12:44:18] INFO: switch from "192.168.1.101" to "MY-PC1" at -1909,653
[2018-06-17T12:44:18] INFO: entering screen
[2018-06-17T12:44:18] INFO: switch from "MY-PC1" to "192.168.1.101" at 1919,655
[2018-06-17T12:44:18] INFO: leaving screen
[2018-06-17T12:44:18] DEBUG: open clipboard
[2018-06-17T12:44:18] DEBUG: close clipboard
[2018-06-17T12:44:18] DEBUG: ignored screen "MY-PC1" update of clipboard 0 (unchanged)
[2018-06-17T12:44:18] DEBUG: open clipboard
[2018-06-17T12:44:18] DEBUG: close clipboard
[2018-06-17T12:44:18] DEBUG: ignored screen "MY-PC1" update of clipboard 1 (unchanged)
[2018-06-17T12:44:18] DEBUG: dropped bogus delta motion: -2902,+64
[2018-06-17T12:44:18] INFO: switch from "192.168.1.101" to "MY-PC1" at -1919,655
[2018-06-17T12:44:18] INFO: entering screen
```

**Comment by Mixxus** [comment]
A workaround for this problem is to reduce the polling rate of the used mouse. I am not running into this issue anymore when setting the polling rate down to 125Hz from 1000Hz.

Seems Synergy can't handle high polling rates ...

**Comment by crhendo** [comment]
I had the same problem with my new Logitech Pro Wireless mouse. With a ton of options to play with, I finally discovered that if I reduced REPORT RATE (PER SECOND) from the default of 1000 to the lowest setting of 125 the cursor jump problem disappears. 

Clearly there is something to do with "competing for resources" that requires attention.

**Comment by kimpurcell** [comment]
I'm having the same issue as well. Has this been resolved? PC as the Server and Mac on the right as the Client. Cursor just jumps back and forth. I had to really roll my mouse hard to get it unstuck.

**Comment by ProtoxiDe22** [comment]
@kimpurcell the problem is still there, i'm facing the exact same problem that can be kinda fixed by reducing the polling rate of the mouse

**Comment by figadore** [comment]
I'm so glad someone found a workaround. I was able to get back to normal by switching my mouse from "performance" (high dpi) to "endurance" (low dpi), though I'd like this to be solved so I don't have to switch

**Comment by unixunion** [comment]
I struggle to switch from my Windows(host) to mac(client), as synergy just seems to switch back and forth in a loop and its really sticky to get it to snap into the mac. From mac to Windows is ok. Tried all DPI settings, but nothing makes this work.

4K host(windows)
1440P client(mac)
high res mouse

**Comment by ProtoxiDe22** [comment]
@unixunion you don't have to tweak the DPI, you have to tweak the polling rate of your mouse, these are different things

**Comment by DreadPirateRobertson** [comment]
Hitting this issue as well between a Windows 10 laptop and a Mac OS laptop. I checked on `synergys.exe` Compatibility settings for All Users, and found disabling 'Override high DPI scaling behavior', and disabling Windows 7 Compatibility mode resolved this issue.

**Comment by kimpurcell** [comment]
Someone here mentioned changing the synergys.exe Compatibility settings for All Users by disabling 'Override high DPI scaling behavior', and disabling Windows 7 Compatibility mode resolved this issue. 

I did that and it actually resolved the issue for me too! However, whenever Synergy launches, it sets those settings back to the way it was. So annoying!!! Can anyone tell Symless to stop being annoying? lol

**Comment by kimpurcell** [comment]
I've noticed that it happens whenever I have images in my Windows clipboard.

**Comment by DreadPirateRobertson** [comment]
@kimpurcell I had originally assumed this was the fix, but it turns out simply having the main Synergy window active 'solves' the issue, again pointing to resource restraints. I eventually just turned down the polling rate of my mouse to 250 which cleared up the majority of the issue.

**Comment by adamroach** [comment]
I'm seeing this as well. I can verify that it seems to go away if I populate the clipboard from the mac, and then reappears if I populate the clipboard from Windows.

**Comment by kimpurcell** [comment]
> I'm seeing this as well. I can verify that it seems to go away if I populate the clipboard from the mac, and then reappears if I populate the clipboard from Windows.

yup, that's exactly what happens. this "bug" has been there forever, never fixed.

**Comment by richard-mah** [comment]
[synergy.log](https://github.com/symless/synergy-core/files/5141979/synergy.log)
Can't believe this bug still exists on a paid product. I have to deal with this annoying issue so many times per day during my work day I'm about ready to go back to having multiple keyboards and mice on my desk. Imagine if the devs and the owner of Synergy used this at work and at home all the time. They'd have fixed it by now!

Having to fight with the computer for 5-15s to drag your mouse from one screen to the next. First you drag quickly, then slowly, then all sorts of different speeds, finally you just randomly and furiously wiggle your mouse left and right as fast as you can hoping it will move to the next screen.

I'll attach my log file where I startup with DEBUG2 level logging, and I just move my mouse from MSI (Server) on the left to AUSYD1LT006 (Client) on the right, then back again.
[synergy.zip](https://github.com/symless/synergy-core/files/5141988/synergy.zip)

**Comment by frostyfrog** [comment]
I will admit that yesterday I saw an interesting variation of this bug that I'd never seen before. I dragged my mouse to the edge of my monitor and it reset the mouse to the center of the screen.

And yeah, I'll agree with with you Richard, This bug has existed since before it became a paid product back when it was donation ware and you could download and use it for free.

**Comment by NateroniPizza** [comment]
Recent purchaser - this is my main complaint with this software. If I move it slowly, it goes fine. If I click it across my other two screens to get to the other PC, it stops dead in its tracks at the border. No issue going from client to server.

**Comment by thallada** [comment]
Setting my mouse reporting rate to 125 in Logitech G Hub completely eliminates the issue for me, but I hate using a mouse at that report rate. It just feels... wrong.

At a report rate of 1000 I can make it across the screen if I go *really* slow, but even then I can see the mouse flash multiple times in the center of the host screen before it makes it to the client. There is obviously something wrong with how Synergy is getting the mouse position.

The clipboard contents have no effect on the issue for me. Nor does having the synergy window active. My host is Windows 10, client is mac.

I use synergy every day and I can't express how much frustration this bug has caused me over the past year or so. I literally have to fight against this software every time I want to switch screens. "Simply move your mouse between your computers effortlessly" the Synergy site says. Ha, okay. I would switch to an alternative immediately but unfortunately I don't know of any.

**Comment by mog3n** [comment]
> Setting my mouse reporting rate to 125 in Logitech G Hub completely eliminates the issue for me, but I hate using a mouse at that report rate. It just feels... wrong.
> 
> At a report rate of 1000 I can make it across the screen if I go _really_ slow, but even then I can see the mouse flash multiple times in the center of the host screen before it makes it to the client. There is obviously something wrong with how Synergy is getting the mouse position.
> 
> The clipboard contents have no effect on the issue for me. Nor does having the synergy window active. My host is Windows 10, client is mac.
> 
> I use synergy every day and I can't express how much frustration this bug has caused me over the past year or so. I literally have to fight against this software every time I want to switch screens. "Simply move your mouse between your computers effortlessly" the Synergy site says. Ha, okay. I would switch to an alternative immediately but unfortunately I don't know of any.

Exactly happening to me as well. Frustrating.

**Comment by domenic** [comment]
I'll just chime in to confirm that this appears to be mouse report rate-related. I set mine from 1000 to 250 in Logitech G HUB and the problem is now better.

**Comment by kimpurcell** [comment]
Symless is hiring. Whoever is applying for the developer position, this could be a mega booster if you could fix it! 😊

**Comment by renoir789** [comment]
Have the same problem on Windows 10 server and Ubuntu Client. Moving across the boundary slowly works. Trying to move across normally doesn't. Changing the polling rate on the mouse directly doesn't seem to fix it for me.

**Comment by kimpurcell** [comment]
I gave up on this and just extended the monitor to my PC. No longer doing the mac crossover thing, too much of a struggle with Synergy. Doing a Teamviewer full-screen session is easier than working through these bugs. Used a LAN cable though, latency is mad low. Works for my use case.

**Comment by jbratu** [comment]
I have two windows 10 systems with 4k monitors and had the same issue - crossing the boundary slowly works but moving quickly causes the mouse to bounce back and force near the edge of the screen before going into the next screen. I couldn't adjust the mouse polling rate, the relative position setting didn't work, and all DPI scaling is at 100%. I'm going to try another solution for a while.

**Comment by Wavesonics** [comment]
Ya I have the exact same issue @jbratu describes

**Comment by JungleM0nkey** [comment]
Same issue. Switching my logitech mouse to 125 polling rate fixes the problem but I hate using my mouse like this

**Comment by MrWedge18** [comment]
I had the same issue, but after a restart (that triggered a windows update and a ghub update) the issue seems to have gone away.

**Comment by Erid** [comment]
Is there anything we can do to help in getting this issue resolved?, it's very annoying.

**Comment by jbratu** [comment]
Since January I've been using Multiplicity from Stardock and it's worked solidly since switching. I've been a paid Synergy/Symless user for a long time and I like the tool but for my daily use this alternative has been better. :-S

**Comment by Kage-Yami** [comment]
While still not perfect, I've found that on my server (Windows 11 Insiders), enabling "Don't take foreground window on Windows servers" improves things a bit (coupled with an Arch Linux client to the left of the primary monitor).

**Comment by kimpurcell** [comment]
> While still not perfect, I've found that on my server (Windows 11 Insiders), enabling "Don't take foreground window on Windows servers" improves things a bit (coupled with an Arch Linux client to the left of the primary monitor).

I gave up and started using TeamViewer instead, with a LAN cable connection. It's free, really snappy and much more flexible than Synergy. No mouse issues or anything.

**Comment by levitanong** [comment]
I have a similar setup with most of the folks here. Windows server, mac client. I'm on a 120Hz monitor, 125 Hz polling mouse polling rate.

Strangely, the issue seems much worse for me. Even if I move my mouse slowly to "cross over" to the client, the mouse gets rebuffed a few times before being "allowed" in.

**Comment by VanixLH** [comment]
Shocked that this is still an issue on a paid product with no developer response for years, I am now really questioning my support for this software. I have this issue on my Logitech G502 mouse when I use Synergy between my Windows 11 desktop and my Macbook Pro on Monterey, I basically cannot move the mouse over unless I stop at the border and move it over slowly. Works fine on a random Dell bluetooth mouse my work gave me, as others mentioned many times it seems like it's a polling issue.

This is like, the one thing you are advertising your software can do. What is the point if it only works with specific mouse settings under specific circumstances?

**Comment by nickfmc** [comment]
> Shocked that this is still an issue on a paid product with no developer response for years, I am now really questioning my support for this software. I have this issue on my Logitech G502 mouse when I use Synergy between my Windows 11 desktop and my Macbook Pro on Monterey, I basically cannot move the mouse over unless I stop at the border and move it over slowly. Works fine on a random Dell bluetooth mouse my work gave me, as others mentioned many times it seems like it's a polling issue.
> 
> This is like, the one thing you are advertising your software can do. What is the point if it only works with specific mouse settings under specific circumstances?

does it work if you lower your Poling rate on your G502? it should. Unless you are an FPS gamer you won't notice a difference going down to a lower polling rate.

**Comment by leifcr** [comment]
Same issue here with two Windows machines. Happens quite often when the 'slave' requests to elevate, or when logging in from a lock screen

**Comment by milesturin** [comment]
Wow, this is actually causing me to regret this purchase. This needs a fix ASAP.

**Comment by nickfmc** [comment]
> Wow, this is actually causing me to regret this purchase. This needs a fix ASAP.

after seeing your message I was wondering how often they actually update the software, as my version was OLD, went to my login and realized the software doesn't update itself! now I'm actually seeing HUGE improvements with the G602 in the latest release, but Synergy never checks itself for updates, so make sure to manually check for updates every so often! and get version 1.14.3 I'd be interested if anyone else has seen improvements with this version.

**Comment by Wavesonics** [comment]
same, I just updated tanks to your message and it all seems fixed now!

**Comment by nickfmc** [comment]
Can we get an admin to link all of these issues together?? 
https://github.com/symless/synergy-core/issues/6042
https://github.com/symless/synergy-core/issues/4911

I did a bunch of testing with 3 mice and poling rates and wired vs. wifi connection. What I have found is that with version 1.14.3 of Synergy, on a wired connection up to 500hz is almost undetectable now so thats good, however on wifi I have to go down to 125 for a clean transition.  I also wonder how much of this issue if any has to do with the polling rate matching on the 2 devices you are moving between. I''m using windows and Ubuntu and am going to do more testing on matching poling rates to see if that is a factor.

**Comment by alexander-cato** [comment]
**I found that enabling "Switch after waiting" to any degree, even down to the lowest settable value of 10ms has fixed this issue for me reliably.** 

It seems that the client and server need a few microseconds to communicate with one another when switching. Synergy is just failing the connection and jumping the cursor back to the server display after an unreasonably short period if that communication takes too long. 

Perhaps this short switch after waiting setting should be enabled by default if Synergy cannot figure out how to make it work with it disabled by extending the timeout duration slightly.

**Comment by eXo0ne** [comment]
> **I found that enabling "Switch after waiting" to any degree, even down to the lowest settable value of 10ms has fixed this issue for me reliably.**

Yup, for me too! I set it down to 10ms and it's perfect. I might go 25ms just to give it a little bit extra time, it's completely unnoticeable.

**Comment by scottgreenup** [comment]
Just adding another user to the pool.

My issue more exactly.

- I have two MacBook M1s on two seperate 4K screens at the same DPI and the same refresh rate.
- One MacBook is plugged in via HDMI; it's the client.
- One MacBook is plugged in via USB-C/DisplayPort; it's the server.
- If I move from server to client, flawless everytime.
- If I move from client to server, my mouse sometimes be flawless, sometimes it will start in the center of the screen.
- I never see any stuttering or flashing.

I will note that there seems to be some network lag, I am over WiFi and sometimes the input to my client will stutter. I'm wondering if the tolerance for that is what is causing the jump.

I've tried what @alexander-cato mentioned (https://github.com/symless/synergy-core/issues/5992#issuecomment-1081073648) but that didn't help. I tried it from 10ms, to 100ms, and even 1000ms; it would still happen randomly.

Both computers are running 1.14.5-stable-1b7140f2.

If I managed to find two dongles to try using cat-5 instead of WiFi, I'll report back.

**Comment by thallada** [comment]
I have switched to https://www.sharemouse.com/ and it is working much better for me. It does not have this mouse report rate issue (and the mouse speed can even be configured easily with a slider).

**Comment by mijagourlay** [comment]
I'm having the same issue between a Win10 machines with 3 monitors side-by-side (which is the server), and a MacBook Pro 16 (client), and a Logitech G 700s.

Synergy version 1.14.6-snapshot-88fdd263

I tried changing the mouse report rate (I tried all settings, including 125, 200, 250, 500, 1000) and various DPI settings.

I also tried changing "Switch after waiting" to various settings and that seemed to make it worse, maybe.

**Comment by nickfmc** [comment]
@mijagourlay are all of your devices on wired connection, if not which ones are on wifi? wifi is a big variable with this issue, so its helpful to include that info.

**Comment by desymless** [comment]
> @mijagourlay are all of your devices on wired connection, if not which ones are on wifi? wifi is a big variable with this issue, so its helpful to include that info.

Are you thinking that perhaps the latency of a wifi connection is causing the bouncy transition issue?

**Comment by nickfmc** [comment]
> > @mijagourlay are all of your devices on wired connection, if not which ones are on wifi? wifi is a big variable with this issue, so its helpful to include that info.
> 
> Are you thinking that perhaps the latency of a wifi connection is causing the bouncy transition issue?

I did a LOT of testing with 4 mice, 3 computers , win/ubuntu/mac and wired/wireless connections. and I found that with a mouse with adjustable polling rate set to it's minimum and a wired connection I could make the issue completely go away. on wifi I could not 100% stop the issue with adjusting the poling rate no matter what I did, and mice without adjustable poling rates on wifi were the worst condition by far! however the connection method was 100% a factor in my testing so I think it's important to know how peoples devices are connected to the network so we can see trends in people having issues.
